cloud/vmware/vmware_cluster.py

```diff
@@ -0,0 +1,286 @@
+#!/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Joseph Callen <jcallen () csc.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: vmware_cluster
+short_description: Create VMware vSphere Cluster
+description:
+    - Create VMware vSphere Cluster
+version_added: 2.0
+author: Joseph Callen
+notes:
+requirements:
+    - Tested on ESXi 5.5
+    - PyVmomi installed
+options:
+    hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter
+        required: True
+        version_added: 2.0
+    username:
+        description:
+            - The username of the vSphere vCenter
+        required: True
+        aliases: ['user', 'admin']
+        version_added: 2.0
+    password:
+        description:
+            - The password of the vSphere vCenter
+        required: True
+        aliases: ['pass', 'pwd']
+        version_added: 2.0
+    datacenter_name:
+        description:
+            - The name of the datacenter the cluster will be created in.
+        required: True
+        version_added: 2.0
+    cluster_name:
+        description:
+            - The name of the cluster that will be created
+        required: True
+        version_added: 2.0
```

Comment:
for the common options for the vmware modules, we can move the docs to a 'doc fragment' and share it across modules, as we do with code.